### PR TITLE
HUB-11 IDP journey hint

### PR DESCRIPTION
**Pull request by swilson96**
If we have a journey hint cookie, modify the sign-in page to display that IDP prominently as per the design on the ticket.

Changes to piwik events:
* New event fired whenever we show a hint "Sign In Journey Hint Shown - ${idp_name}"
* Existing sign in event has " - Hint Followed" appended if the user followed a journey hint
* Existing sign in event has " - Hint Ignored" appended if the user ignored a journey hint

In the non-JS case we need a new session param to tell the idp redirect controller to send the correct event type - that controller can't check the hint cookie itself because the cookie has already been overwritten to the new IDP choice that was just clicked

### CODE REVIEW

File: app/views/sign_in/index.html.erb [hunk 1]
```diff
@@ -6,11 +6,27 @@
 <div class="grid-row">
   <div class="column-two-thirds">
     <h1 class="heading-large"><%= t 'hub.signin.heading' %></h1>
-    <p><%= t 'hub.signin.registration_message_html', href: link_to(t('hub.signin.about_link'), begin_registration_path, id: 'begin-registration-route') %></p>
+    <p class="lede"><%= t 'hub.signin.registration_message_html', href: link_to(t('hub.signin.about_link'), begin_registration_path, id: 'begin-registration-route') %></p>
```
[the remark this answers is not in the document]
Comment (swilson96) [reply app/views/sign_in/index.html.erb]: Yes, it's important for the layout. http://govuk-elements.herokuapp.com/typography/#typography-lead-paragraph

[the remark this answers is not in the document]
Comment (swilson96) [reply app/views/sign_in/index.html.erb]: It looked like nonsense to me too but it is in fact a word: https://www.merriam-webster.com/dictionary/lede

File: app/views/sign_in/index.html.erb [hunk 2]
```diff
@@ -6,11 +6,27 @@
...
 </div>
 
-<div class="grid-row js-continue-to-idp" data-location="<%= url_for(controller: 'sign_in', action: 'select_idp_ajax', locale: I18n.locale)  %>">
+<% if defined?(@suggested_idp) && !@suggested_idp.empty? %>
+  <div class="grid-row">
+    <div class="column-two-thirds">
+      <p><%= t 'hub.signin.last_certified', company_name: @suggested_idp[0].display_name %></p>
+    </div>
+  </div>
+  <div class="grid-row js-continue-to-idp journey-hint" data-location="<%= url_for(controller: 'sign_in', action: 'select_idp_ajax', locale: I18n.locale) %>">
```
Comment (jakubmiarka) [inline app/views/sign_in/index.html.erb]: and the journey-hint class of the div? I can't see any usage of it.

Comment (swilson96) [reply app/views/sign_in/index.html.erb]: No there's no usage of that - I was potentially going to use it for a web test to differentiate between the hinted IDP and the same IDP in the big list. I'll remove it

**Pull request merged**